src/index.js

```diff
@@ -99,6 +100,8 @@ export default function nodeResolve ( options = {} ) {
 				id = resolve( importer, '..', importee );
 			}
 
+			if (only && !only.some(pattern => id.match(pattern))) return null;
```

Comment:
For performance reasons it might also make sense to initially convert all entries to RegExp (in L.47 where it will only happen once) and then use `pattern.test(id)` instead (which will make all invocations faster, cf. e.g. https://stackoverflow.com/a/10940138)